Mathlib/GroupTheory/FreeGroup/CyclicallyReduced.lean

```diff
@@ -200,4 +206,79 @@ theorem reduce_flatten_replicate (h : IsReduced L) (n : ℕ) :
   | n + 1 => reduce_flatten_replicate_succ h n
 
 end reduceCyclically
+
+section pow_left_inj
+variable [DecidableEq α]
+open reduceCyclically
+
+@[to_additive FreeAddGroup.nsmul_right_inj]
+theorem pow_left_inj {x y : FreeGroup α} {n : ℕ} (hn : n ≠ 0) : x ^ n = y ^ n ↔ x = y := by
```

Comment:
I found this proof pretty dense, especially given the size of the hypotheses you're building. I don't have concrete suggestions